workspaces/core/src/search/private/components/Search.tsx

```diff
@@ -0,0 +1,235 @@
+  page?: string;
+  backdrop?: ReactNode;
+  className?: string;
+}
+
+export function Search({
+  selected: selectedParent,
+  placeholder = "Search",
+  onChangeSelected,
+  onChangeSearch,
+  onChangeOpen,
+  renderLink,
+  children,
+  isOpen,
+  search,
+  page,
+  className,
+}: SearchProps) {
+  const inputRef = useRef<MutableRefObject<HTMLInputElement>>(null);
+  const [ref] = useClickOutside();
+
+  const [selected, setSelected] =
+    typeof selectedParent === "number" && onChangeSelected
+      ? [selectedParent, onChangeSelected]
+      : useState<number>(0);
+
+  const [searchPrefix, setSearchPrefix] = useState<string[] | undefined>();
+
+  function handleChangeSelected(direction?: "up" | "down") {
+    const items = document.querySelectorAll(".moon-search-list-item");
+
+    let index = 0;
+    let newIndex = 0;
+    let newItem: Element;
+
+    if (direction === "down") {
+      items.forEach((_, i) => {
+        if (i === selected) {
+          index = i;
+        }
+      });
+
+      newIndex = index === items.length - 1 ? 0 : index + 1;
+    } else if (direction === "up") {
+      items.forEach((_, i) => {
+        if (i === selected) {
+          index = i;
+        }
+      });
+
+      newIndex = !index ? items.length - 1 : index - 1;
+    } else {
+      setSelected(0);
+    }
+
+    newItem = items[newIndex];
+
+    if (newItem && typeof newIndex === "number") {
+      setSelected(newIndex);
+      newItem.scrollIntoView({
+        behavior: "smooth",
+        block: newIndex ? "center" : "end",
+      });
+    }
+  }
+
+  function handleSelect() {
+    const items = document.querySelectorAll(
+      ".moon-search-list-item"
+    ) as NodeListOf<HTMLButtonElement | HTMLAnchorElement>;
+
+    let index = 0;
+    let item: HTMLAnchorElement | HTMLButtonElement;
+
+    items.forEach((_, i) => {
+      if (i === selected) {
+        index = i;
+      }
+    });
+
+    item = items[index];
+
+    if (item) {
+      item.click();
+
+      if (
+        item.attributes.getNamedItem("data-close-on-select")?.value === "true"
+      ) {
+        onChangeOpen(false);
+      }
+    }
+  }
+
+  useEffect(() => {
+    handleChangeSelected();
+  }, [search]);
+
+  useEffect(() => {
+    setSelected(0);
+  }, [page]);
+
+  return (
+    <div
+      ref={ref}
+      onKeyDown={(e) => {
+        if (
+          e.key === "ArrowDown" ||
+          (e.ctrlKey && e.key === "n") ||
+          (e.ctrlKey && e.key === "j")
+        ) {
+          e.preventDefault();
+          e.stopPropagation();
+          handleChangeSelected("down");
+        } else if (
+          e.key === "ArrowUp" ||
+          (e.ctrlKey && e.key === "p") ||
+          (e.ctrlKey && e.key === "k")
+        ) {
+          e.preventDefault();
+          e.stopPropagation();
+          handleChangeSelected("up");
+        } else if (e.key === "Enter") {
+          e.preventDefault();
+          e.stopPropagation();
+          handleSelect();
+        }
+      }}
```

Comment:
Can we please move it to the separate onKeyDown handler function, I think it makes the code a bit clear